app/models/timeline_entry.rb

```diff
@@ -2,4 +2,14 @@ class TimelineEntry < ApplicationRecord
   belongs_to :coronavirus_page
   validates :heading, presence: true, length: { maximum: 255 }
   validates :content, presence: true
+  after_create :set_position
+
+  def set_position
+    coronavirus_page.timeline_entries.each do |timeline_entry|
+      new_position = timeline_entry.position.present? ? timeline_entry.position + 1 : 1
+      timeline_entry.update!(position: new_position)
+    end
+
+    update!(position: 1)
+  end
```

Comment:
There's an aspect of dubiousness that we briefly persist an invalid entry into the database by allowing nil. I wonder if this might be better as a before_save block where you set the position as 1 and update_all the other ones to be one greater? It has a small risk of not being in a transaction that you wouldn't necessarily have the item persisted, but then you have a similar risk in an after_create that the callback falls.

Reply:
Ah sorry I meant `before_create`

Reply:
If I switched to using a `before_create`, `set_position` could be changed to:

```
  def set_position
    coronavirus_page.timeline_entries.update_all("position = position + 1")
    self.position = 1
  end
```

Is that what you're thinking? And then I can switch the position column back to not accepting nils.

Reply:
Yup - that's it. Hopefully I've not missed something that makes that renders that untenable.